src/views/Referrals/Referrals.js

```diff
@@ -124,13 +138,61 @@ export default function Referral(props) {
       });
   };
 
+  const merkleDistributor = getContract(chainId, "MerkleDistributor");
+  const merkleDistributorReader = getContract(chainId, "MerkleDistributorReader");
+
   function handleClaim() {
-    // TODO handle claim
+    console.log("Claiming rewards");
+    setIsClaiming(true);
+    // helperToast.error("Claiming rewards is currently disabled");
+    trackAction("Button clicked", {
+      buttonName: "Claim rewards",
+    });
+    console.log(userProof);
+    let error;
+    if (selectedRound === "latest") {
+      helperToast.error("Cannot claim rewards before round has ended");
+      error = true;
+    }
+    if (!userProof) {
+      helperToast.error("Fetching merkle proof");
```

Comment:
This error message could be a bit more helpful.  Maybe "Fetching merkle proof.  Please wait a minute and try again"